### PR TITLE
Fix unreferenced-pruning bug in unnest

### PR DESCRIPTION
**Pull request by haozhun**
#2535

### CODE REVIEW

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 1]
```diff
@@ -134,6 +134,8 @@ public void testUnnest()
         assertQuery("SELECT a FROM UNNEST(ARRAY[1, 2, 3]) t(a)", "SELECT * FROM VALUES (1), (2), (3)");
         assertQuery("SELECT a, b FROM UNNEST(ARRAY[1, 2], ARRAY[3, 4]) t(a, b)", "SELECT * FROM VALUES (1, 3), (2, 4)");
         assertQuery("SELECT a, b FROM UNNEST(ARRAY[1, 2, 3], ARRAY[4, 5]) t(a, b)", "SELECT * FROM VALUES (1, 4), (2, 5), (3, NULL)");
+        assertQuery("SELECT a FROM UNNEST(ARRAY[1, 2, 3], ARRAY[4, 5]) t(a, b)", "SELECT * FROM VALUES 1, 2, 3");
+        assertQuery("SELECT b FROM UNNEST(ARRAY[1, 2, 3], ARRAY[4, 5]) t(a, b)", "SELECT * FROM VALUES 4, 5, NULL"); // Issue #2536
```
Comment (cberner) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: Remove comment at end, we don't usually document which issue a test case is for

Comment (cberner) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: Add a test case like this, just to make sure we don't have other pruning bugs:
```
SELECT c
FROM UNNEST(ARRAY[1, 2, 3], ARRAY[4, 5]) t(a, b) 
CROSS JOIN (values (8), (9)) t2(c);
```

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 2]
```diff
@@ -134,6 +134,8 @@ public void testUnnest()
         assertQuery("SELECT a FROM UNNEST(ARRAY[1, 2, 3]) t(a)", "SELECT * FROM VALUES (1), (2), (3)");
         assertQuery("SELECT a, b FROM UNNEST(ARRAY[1, 2], ARRAY[3, 4]) t(a, b)", "SELECT * FROM VALUES (1, 3), (2, 4)");
         assertQuery("SELECT a, b FROM UNNEST(ARRAY[1, 2, 3], ARRAY[4, 5]) t(a, b)", "SELECT * FROM VALUES (1, 4), (2, 5), (3, NULL)");
+        assertQuery("SELECT a FROM UNNEST(ARRAY[1, 2, 3], ARRAY[4, 5]) t(a, b)", "SELECT * FROM VALUES 1, 2, 3");
+        assertQuery("SELECT b FROM UNNEST(ARRAY[1, 2, 3], ARRAY[4, 5]) t(a, b)", "SELECT * FROM VALUES 4, 5, NULL");
```
Comment (cberner) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: Add a test case like this, to be sure we don't have other pruning bugs:

```
SELECT c
FROM UNNEST(ARRAY[1, 2, 3], ARRAY[4, 5]) t(a, b) 
CROSS JOIN (values (8), (9)) t2(c);
```

**Comment by cberner** [comment]
commit message is too long. Otherwise looks good

**Comment by haozhun** [comment]
Code changed substantially. Please review again.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java [hunk 3]
```diff
@@ -341,13 +341,7 @@ public PlanNode visitUnnest(UnnestNode node, RewriteContext<Set<Symbol>> context
...
-            for (Map.Entry<Symbol, List<Symbol>> entry : node.getUnnestSymbols().entrySet()) {
-                if (Iterables.any(entry.getValue(), in(context.get()))) {
-                    builder.put(entry);
-                }
-            }
-            Map<Symbol, List<Symbol>> unnestSymbols = builder.build();
+            Map<Symbol, List<Symbol>> unnestSymbols = node.getUnnestSymbols();
             ImmutableSet.Builder<Symbol> expectedInputs = ImmutableSet.<Symbol>builder()
                     .addAll(replicateSymbols)
                     .addAll(unnestSymbols.keySet());
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java]: Remove the `if (unnestSymbols.isEmpty())` also

**Comment by cberner** [comment]
looks good, besides the one comment

**Pull request merged**